src/giskardpy/plugin_world_visualization.py

```diff
@@ -0,0 +1,62 @@
+import hashlib
+
+import py_trees
+import rospy
+from visualization_msgs.msg import Marker, MarkerArray
+
+from plugin import GiskardBehavior
+import giskardpy.identifier as identifier
+from giskardpy.tfwrapper import lookup_pose, get_full_frame_name
+
+
+class WorldVisualizationBehavior(GiskardBehavior):
+    def __init__(self, name, ensure_publish=False):
+        super(WorldVisualizationBehavior, self).__init__(name)
+        self.map_frame = self.get_god_map().get_data(identifier.map_frame)
+        self.marker_namespace = u'planning_world_visualization'
+        self.ensure_publish = ensure_publish
+
+    def setup(self, timeout):
+        self.publisher = rospy.Publisher(u'~visualization_marker_array', MarkerArray, queue_size=1)
+        self.ids = set()
+        return super(WorldVisualizationBehavior, self).setup(timeout)
+
+    def update(self):
+        markers = []
+        objects = self.get_world().get_objects().values()
+        for object in objects:
+            for link_name in object.get_link_names():
+                if object.has_link_visuals(link_name):
+                    marker = object.link_as_marker(link_name)
+                    if marker is None:
+                        continue
+                    marker.header.frame_id = self.map_frame
+                    marker.id = int(hashlib.md5(object.get_name() + link_name).hexdigest()[:6],
+                                    16)  # FIXME find a better way to give the same link the same id
```

Comment:
You need to encode the string as `utf-8` like in the visualization plugin. https://github.com/SemRoCo/giskardpy/blob/6cb571ea0533e6adb37562c809d192b11a180ae2/src/giskardpy/plugin_visualization.py#L35
Otherwise it doesn't work with python 3.

Reply:
Also you should get used to concatenate strings like this: `'{}{}'.format(object.get_name(), link_name)`. It won't make a significant difference here, but its generally faster and more readable in complex cases.